Mathlib/GroupTheory/Submonoid/ZeroDivisors.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.GroupTheory.Submonoid.Basic
+
+/-!
+# Zero divisors
+
+## Main definitions
+
+ * `nonZeroDivisorsLeft`: the elements of a `MonoidWithZero` that are not left zero divisors.
+ * `nonZeroDivisorsRight`: the elements of a `MonoidWithZero` that are not right zero divisors.
+
+-/
+
+section MonoidWithZero
+
+variable (M₀ : Type _) [MonoidWithZero M₀]
```

Comment:
Technically, this probably makes sense for `SemigroupWithZero` and `Subsemigroup` as well...

Reply:
It does but I need the `Submonoid` type since I want to use `Submonoid.pow_mem` (which cannot exist for `Subsemigroup` because the power might be zero).

We could of course make both definitions but I propose waiting since we do not have an application in mind.

Reply:
Sure, it just got me thinking about whether I had ever encountered a ring without unit and with nilpotent elements, but I could not come up with one.  Likely, because I do not often see rings without units!  :smile: